waveform_modes.py

```diff
@@ -315,6 +315,77 @@ def indices(self, ell_m):
             raise ValueError("Input `ell_m` should be an Nx2 sequence of integers")
         return ell_m[:, 0] * (ell_m[:, 0] + 1) - self.ell_min ** 2 + ell_m[:, 1]
 
+
+    def inner_product(self, b, allow_LM_differ=False, allow_times_differ=False):
+        """Compute the all-angles inner product <self, b>.
+        self and b should have the same spin-weight, set of (ell,m)
+        modes, and times.  Differing sets of modes and times can
+        optionally be turned on.
+
+        Parameters
+        ----------
+        b : WaveformModes object
+            The other set of modes to inner product with.
+
+        allow_LM_differ : bool, optional [default: False]
+            If True and if the set of (ell,m) modes between self and b
+            differ, then the inner product will be computed using the
+            intersection of the set of modes.
+
+        allow_times_differ: bool, optional [default: False]
+            If True and if the set of times between self and b differ,
+            then both WaveformModes will be interpolated to the
+            intersection of the set of times.
+
+        Returns
+        -------
+        inner_product : complex
+            <self, b>
+        """
+
+        from .mode_calculations import inner_product
+        from .extrapolation import intersection
+
+        if (self.spin_weight != b.spin_weight):
+            raise ValueError("Spin weights must match in inner_product")
+
+        LM_clip = None
+        if ((self.ell_min != b.ell_min) or (self.ell_max != b.ell_max)):
+            if (allow_LM_differ):
+                LM_clip = slice( max(self.ell_min, b.ell_min),
+                                 min(self.ell_max, b.ell_max) + 1 )
+                if (LM_clip.start >= LM_clip.stop):
+                    raise ValueError("Intersection of (ell,m) modes is "
+                                     "empty.  Assuming this is not desired.")
+            else:
+                raise ValueError("ell_min and ell_max must match in inner_product "
+                                 "(use allow_LM_differ=True to override)")
+
+        t_clip = None
+        if ((self.t.shape != b.t.shape) or (self.t != b.t).any()):
```

Comment:
Just FYI: this is basically what `np.array_equal` does.

Reply:
Done.